sunpy/net/dataretriever/sources/gong.py

```diff
@@ -0,0 +1,282 @@
+        instruments = ['bb', 'ct', 'le', 'ml', 'td', 'ud', 'z'] #for Magnetogram and intensity
+        chk_instr, chk_physobs = 0, 0
+        chk_wavelength = 0
+        values = [6562, 6563, 6768]
+        for x in query:
+            if x.__class__.__name__ == 'Instrument' and x.value.lower() in instruments:
+                chk_instr += 1
+            if x.__class__.__name__ == 'Physobs' and x.value in physobs:
+                chk_physobs += 1
+            if (x.__class__.__name__ == 'Wavelength' and int(x.min.value) in values and int(x.max.value) in values and (x.unit.name).lower()=='angstrom'):
+                chk_wavelength += 1
+        if chk_instr == 1 or chk_physobs == 1:
+            return True
+        return False
+
+
+class FARSIDEClient(GenericClient):
+    """
+    Returns a list of URLS to FARSIDE files corresponding to value of input timerange.
+    URL source: `http://farside.nso.edu/oQR/fqo/`.
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+    
+    Instrument: Fixed argument = 'farside'
+            
+    Returns
+    -------
+    urls: list
+    list of urls corresponding to requested time range.
+    
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+    >>> results = Fido.search(Time('2015/4/2','2015/4/4'), a.Instrument('farside'))
+    >>> print(results)
+    [<Table length=4>
+         Start Time           End Time      Source Instrument
+           str19               str19         str4     str7   
+    ------------------- ------------------- ------ ----------
+    2015-04-02 00:00:00 2015-04-03 00:00:00   GONG    farside
+    2015-04-03 00:00:00 2015-04-04 00:00:00   GONG    farside
+    2015-04-04 00:00:00 2015-04-05 00:00:00   GONG    farside
+    2015-04-05 00:00:00 2015-04-06 00:00:00   GONG    farside]
+    
+    >>> response = Fido.fetch(results)
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        START_DATE = datetime.datetime(2006, 5, 13, 12, 0, 0)
+        if timerange.start < START_DATE:
+            raise ValueError('Earliest date for which FARSIDE data is available is {:%Y-%m-%d}'.format(START_DATE))
+        url_pattern = 'http://farside.nso.edu/oQR/fqo/{:%Y%m}/mrfqo{:%y%m%d}/mrfqo{:%y%m%d}t{:%H%M}.fits'
+        tot_days = (timerange.end - timerange.start).days
+        all_days = timerange.split(tot_days)
+        result = list()
+        for dt in all_days:
+            times = [datetime.datetime(dt.start.year, dt.start.month, dt.start.day, 0, 0),
+                     datetime.datetime(dt.start.year, dt.start.month, dt.start.day, 12, 0)]
+            [result.append(url_pattern.format(dt_, dt_, dt_, dt_)) for dt_ in times]
```

Comment:
you can add a name to the pattern to then used only one `dt_`  in `.format()`.
Also, I don't like the line in this syntax. You could do like I suggested above.